src/main/java/com/mparticle/kits/TaplyticsKit.java

```diff
@@ -0,0 +1,300 @@
+package com.mparticle.kits;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.support.annotation.Keep;
+
+import org.json.JSONException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+import java.math.BigDecimal;
+
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.mparticle.MParticleOptions;
+import com.mparticle.MParticleTask;
+import com.mparticle.commerce.CommerceEvent;
+import com.mparticle.commerce.Product;
+import com.mparticle.commerce.TransactionAttributes;
+import com.mparticle.identity.IdentityApiRequest;
+import com.mparticle.identity.IdentityApiResult;
+
+import com.taplytics.sdk.SessionInfoRetrievedListener;
+import com.taplytics.sdk.Taplytics;
+import com.taplytics.sdk.TaplyticsExperimentsLoadedListener;
+import com.taplytics.sdk.TaplyticsHasUserOptedOutListener;
+
+import org.json.JSONObject;
+
+public class TaplyticsKit extends KitIntegration
+        implements
+        KitIntegration.AttributeListener,
+        KitIntegration.EventListener,
+        KitIntegration.CommerceListener,
+        KitIntegration.ApplicationStateListener {
+
+    /**
+     * Option Keys
+     */
+    private static final String API_KEY = "apiKey";
+    private static final String AGGRESSIVE = "TaplyticsOptionAggressive";
+    private static final String USER_ID = "user_id";
+    private static final String EMAIL = "email";
+
+    /**
+     * Event Keys
+     */
+    private static final String EventAppBackground = "appBackground";
+    private static final String EventAppForeground = "appForeground";
+
+    /**
+     * tlOptions get and set methods
+     */
+
+    private static Map<String, Object> tlOptions;
+
+    public static Map<String, Object> getTlOptions() { return tlOptions; }
+
+    public static void setTlOptions(Map<String, Object> options) {
+        tlOptions = options;
+    }
+
+    private HashMap<String, Object> mergeOptions(Map<String, Object> tlOptions, Map<String, Object> configuration) {
+        HashMap<String, Object> merged = new HashMap<>(configuration);
+        for (Map.Entry<String, Object> entry : tlOptions.entrySet()) {
```

Comment:
if I'm reading this right it will crash by default because `tlOptions` is null? also customers can null it out.